ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -1052,6 +1055,41 @@ func isPIInstanceRefreshFunc(client *instance.IBMPIInstanceClient, id, instanceR
 	}
 }
 
+func isWaitForPIInstanceAvailableOrShutoffAfterUpdate(ctx context.Context, client *instance.IBMPIInstanceClient, id string, timeout time.Duration) (interface{}, error) {
+	log.Printf("Waiting for PIInstance (%s) to be available and active or shutoff ", id)
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Updating, Warning},
+		Target:     []string{State_Active, OK, State_Shutoff},
+		Refresh:    isPIInstanceShutoffOrActiveAfterResourceChange(client, id),
+		Delay:      Timeout_Delay,
+		MinTimeout: 5 * time.Minute,
+		Timeout:    timeout,
+	}
+
+	return stateConf.WaitForStateContext(ctx)
+}
+
+func isPIInstanceShutoffOrActiveAfterResourceChange(client *instance.IBMPIInstanceClient, id string) retry.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		pvm, err := client.Get(id)
+		if err != nil {
+			return nil, "", err
+		}
+		if strings.ToLower(*pvm.Status) == State_Active && pvm.Health.Status == OK {
```

Comment:
You would also need to check the `Arg_HealthStatus`

Reply:
I did not do that since the old function `isWaitforPIInstanceUpdate` (now called `isWaitForPIInstanceShutoffAfterUpdate`) did not include that. Should I add that to both?

Reply:
Added to active block. Kept shutoff the same since the other update function also only checks for "OK" health status when checking for shutdown (keeps behavior the same)